### PR TITLE
Created Circular Queue in C

### PR DESCRIPTION
**Pull request by iamvs-2002**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
Yes, I have read them.
### Description

(Write your answer here.)
Implemented the circular queue data structure in the C language. 
### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#1741 
(Write your answer here.)

### CODE REVIEW

File: C/ds/circularqueue.c [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+            out == false;
+            return 0;
+        }
+        else
+        {
+            out = false;
+            return 0;
+        }
+    }
+}
```
Comment (HarshCasper) [inline C/ds/circularqueue.c]: Add a new line at the end of the file

File: C/ds/circularqueue.c [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    out = st.data[st.front];
+
+    printf("%d -> Removed! \n", out);
+}
+
+int main(void)
+{
+    int user, value;
+    st.front = -1; //setting the front pointer to null
+    st.rear = -1; //setting the rear pointer to null
```
Comment (HarshCasper) [inline C/ds/circularqueue.c]: Keep Comments above the Line of Code. Not beside them.

File: C/ds/circularqueue.c [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    else
+    {
+        st.rear++;
+        st.data[st.rear] = number; //adding the value to the queue
+        printf("%d -> Added! \n", st.data[st.rear]);
+    }
+}
+
+void dequeue()
+// removing a specific node fromt he queue
```
Comment (HarshCasper) [inline C/ds/circularqueue.c]: Typo Errors.

File: C/ds/circularqueue.c [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+{
+    int data[5]; // creating the queu structure with data, front and rear nodes pointing to front and the last(behind) node
+    int front;//a node containg the value of a pointer/address pointing to the front node
+    int rear;//a node containing the value of a pointer/address pointing to the back/behind/rear node
+} mainf;
+
+mainf st;//the name of the structure
+//we will be referencing our structure with st in the code below
+
+//In the circular queue, in the last node, the front pointer points to the first node.
```
Comment (HarshCasper) [inline C/ds/circularqueue.c]: If you are creating multiple comments, prefer to use multi-line comment.

File: C/ds/circularqueue.c [hunk 5]
```diff
@@ -0,0 +1,86 @@
+#include <stdio.h> //headers
```
Comment (HarshCasper) [inline C/ds/circularqueue.c]: Add a beginner pitch about your Code.

File: C/ds/circularqueue.c (hunk 1 above)
Comment (HarshCasper) [inline C/ds/circularqueue.c]: Add a sample I/O.

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: C/ds/circularqueue.c (hunk 1 above)
Comment (iamvs-2002) [reply C/ds/circularqueue.c]: Ohhk

File: C/ds/circularqueue.c (hunk 3 above)
Comment (iamvs-2002) [reply C/ds/circularqueue.c]: ooops

**Review by iamvs-2002 — COMMENTED** [review]
Added the comments and resolved the typo errrors.

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Code looks good. Make the apt README changes as well, and it would be ready to merge.

**Comment by iamvs-2002** [comment]
@HarshCasper Done!

**Review by HarshCasper — APPROVED** [review]

**Comment by iamvs-2002** [comment]
@HarshCasper Kindly add the dwoc label as the core team says that the label must be both on the pr and the issue.

**Pull request merged**